### PR TITLE
Add checksum for rdma-core@39.1

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/rdma-core/package.py [hunk 1]
```diff
@@ -50,9 +51,13 @@ def determine_version(cls, lib):
...
-            '-DPYTHON_INCLUDE_DIR={0}'
-            .format(self.spec['python'].headers.directories[0])
+            '-DCMAKE_INSTALL_RUNDIR=/var/run'
         ]
+
+        if self.spec.satisfies('@:39.0'):
+            cmake_args.extend([
+                '-DPYTHON_LIBRARY={0}'.format(self.spec['python'].libs[0]),
+                '-DPYTHON_INCLUDE_DIR={0}'
+                .format(self.spec['python'].headers.directories[0])
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/rdma-core/package.py]:
```suggestion
                self.define('PYTHON_LIBRARY', self.spec['python'].libs[0]),
                self.define('PYTHON_INCLUDE_DIR', self.spec['python'].headers.directories[0])
```

File: var/spack/repos/builtin/packages/rdma-core/package.py [hunk 2]
```diff
@@ -50,9 +51,13 @@ def determine_version(cls, lib):
     def cmake_args(self):
         cmake_args = [
             '-DCMAKE_INSTALL_SYSCONFDIR={0}'.format(self.spec.prefix.etc),
-            '-DCMAKE_INSTALL_RUNDIR=/var/run',
-            '-DPYTHON_LIBRARY={0}'.format(self.spec['python'].libs[0]),
-            '-DPYTHON_INCLUDE_DIR={0}'
-            .format(self.spec['python'].headers.directories[0])
+            '-DCMAKE_INSTALL_RUNDIR=/var/run'
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/rdma-core/package.py:54]: Why is there a hardcoded install dir? This likely isn't compatible with most systems.

Comment (iarspider) [reply var/spack/repos/builtin/packages/rdma-core/package.py:54]: This was added [here](https://github.com/spack/spack/commit/312de47594aa751bddfd20daa341cea4a5e94ebe), maybe @carns can comment?

Comment (carns) [reply var/spack/repos/builtin/packages/rdma-core/package.py:54]: Hi @sethrj and @iarspider,

The reason for this hack/workaround is documented in https://github.com/spack/spack/pull/9370.

The TLDR is that spack has (had?) a tendency to generate very long staging paths, and the rdma-core source code is (was?) constructing static strings from based on those paths.  This led to some compilation problems.  I imagine that there is a better way to solve the problem.

Comment (sethrj) [reply var/spack/repos/builtin/packages/rdma-core/package.py:54]: Sounds sketchy but I guess it's not an issue added by this changeset.

**Review by sethrj — DISMISSED** [review]

**Review by sethrj — APPROVED** [review]

**Pull request merged**